Adding Discrete Vector Field Topology example

finkentA opened the pull request:
Thanks for contributing to TTK!

Before submitting your pull request, please:

- [x] Review our [Contributor Guidelines](https://github.com/topology-tool-kit/ttk-data/blob/dev/CONTRIBUTING.md), in particular regarding the additional entries associated to your example (ParaView state file, Python script, MkDocs documentation, etc.) 

- [x] Please provide a quick description of your example below:

This Pull Request adds an example using the Discrete Vector Field filters recently Pull Requested for TTK. The example uses a 2D vector field dataset(changes.vtu) generated from a piecewise function described in the paper titled "Localized Evaluation for Constructing Discrete Vector Fields". First some noise is added to the vectors using paraview's RandomVectors. Then the VectorWeightCurve filter is applied and adjusted to find a stable region to simplify to(27 discrete critical points). Finally, the TopologicalSkeleton filter is applied with the chosen simplification threshold and visualized with the critical points and 1-separatrices emitted from the saddles(critical dimension=1). The example shows the ability to determine a simplification threshold from the weight curve and visualize the topology of the simplified discrete field.

finkentA committed:
Add files via upload

finkentA committed:
Merge branch 'topology-tool-kit:dev' into dev

finkentA committed:
Add files via upload

finkentA committed:
Update ChangesFlowTopology.md

finkentA committed:
Add files via upload

finkentA committed:
Add files via upload

finkentA committed:
Add files via upload

finkentA committed:
Update README

finkentA committed:
Delete python/ChangesWithNoiseExample.py

finkentA committed:
Discrete Vector Field example

finkentA committed:
Delete states/ChangesWithNoiseTopology.pvsm

finkentA committed:
Discrete Vector Field State example

finkentA committed:
Delete tests/reference_screenshots/ChangesExampleScreenshot.png

finkentA committed:
Discrete Vector Field Screenshot

finkentA committed:
Delete docs/ChangesFlowTopology.md

finkentA committed:
Discrete Vector Field example doc

finkentA committed:
Remove Registration Names

julien-tierny committed:
[DVFT] enhancing state file

julien-tierny commented:
Hi Tanner,
thanks a lot for this great set of PRs.
It all looks already very polished.

One first remark:
I have the impression that the current code has an non-deterministic behavior, at least when it comes to assigning the "Intersecting Manifold" Ids, which can be a problem for testing the code in the continuous integration.

For instance, I slightly updated the state file you provided to show that segmentation with random colors (with `TTKIdentifierRandomizer`, which is deterministic, for a given seed).

However, if you re-execute your filter (by pressing the button `Execute`), you will see that the segmentation colors will change (if you click on `Execute` on `TTKIdentifierRandomizer`, the color will no longer change as this filter is deterministic).

For example:
![Screenshot_20250509_145123](https://github.com/user-attachments/assets/3c0d479c-471d-463d-8520-11f36e0af6df)

After clicking on that button:

![Screenshot_20250509_145148](https://github.com/user-attachments/assets/8ce31d61-1785-4a8d-b96d-2b10e3fe0fb4)

Could you please have a look into this?

julien-tierny committed:
[DVFT] Pass on doc entries

julien-tierny commented:
I confirm that the same issues arises with the provided python scripts: the outputs `Separatrices1.csv` and `Segmentation.vtu` are not deterministic (they change, check with `diff`, from one execution to another).

julien-tierny commented:
(notes for myself: **TODO** black and shorten python script once deterministic).

julien-tierny commented:
Hi Tanner,
one more comment/issue. The input to your example is a VTU file, while it would preferable to have a VTI file (300x300, same number of points). That would simply be more convenient (the file would be much smaller).
However, when I switch to a VTI input, I observe weird patterns on the segmentation (bottom left corner in the screenshot below). As the current code is not deterministic, you may need to hit the button `Execute` multiple times to observe this:
![Screenshot_20250509_152841](https://github.com/user-attachments/assets/381d9c79-5131-4424-9015-0979c64b2f66)

This looks like a bug to me (for a given region, multiple colors appear although there is no separatrices).

julien-tierny committed:
[DVFT] pass on documentation

finkentA commented:
Hi,
Thanks for letting me know about these issues. I believe I have identified  the main issue to be when the code identifies the orbits in parallel. This can potentially generate multiple orbits with various id's and thus changing the results between runs. I will work on fixing this.

finkentA committed:
Merge branch 'topology-tool-kit:dev' into dev

finkentA committed:
Delete changes.vtu

Removing to add the .vti file instead

finkentA committed:
Add changes sampled on an image instead

finkentA committed:
Update README

finkentA committed:
Update discreteVectorFieldTopology.md

finkentA committed:
Update discreteVectorFieldTopology.py

finkentA committed:
Update discreteVectorFieldTopology.pvsm

finkentA commented:
Alright, I believe I have fixed the issue with the orbits, where the orbits will now be uniquely tracked and used in the segmentation, and also updated the input file to be VTI.

julien-tierny committed:
[dvft] fixed io in examples

julien-tierny committed:
[dvft] simplified python example, updated pv state

julien-tierny commented:
Alright, after a couple updates, it looks like this example is ready to be merged (after code merge).

julien-tierny committed:
[dvft] removing ref screenshot (automated)
